formal internal review of AAD baseline

tkol2022 opened the pull request:
# Scope #

This PR is designed to facilitate a formal review of the revised AAD baseline from CISA Scuba principals and the Scuba development team. Following this review the AAD baseline should be in a state for sending to the CISA agency formal review process.

## Note

We have active development issues against this baseline at the moment, including new policy implementations and tweaks to existing policies. If any changes occur during this baseline review which materially affect the ScubaGear code, it may require a notification to the affected developers.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

gdasher commented:
I reviewed the content of the AAD baseline on the emerald branch.  It is approved from my point of view with these changes:

* **MS.AAD.7.\* controls**: Clarify that Microsoft AAD PIM qualifies as a pam system and that a non-PIM PAM may be used if it meets the same set of requirements.  Terminology from PIM is sprinkled throughout these controls (e.g. permanent active), implying that PIM must be used.   Overall, my suggestion is to be explicit that PIM terminology is used through and that another PAM system may be used assuming it meets the requirements, but also that PIM is acceptable to meet the pam requirement in MS.AAD.7.5v1
* (nit) **MS.AAD.4.1v1**: please used fixed width font for the log types so that they stand out from the surrounding text

tkol2022 commented:
@nanda-katikaneni @schrolla @ssatyapal123 Kindly let me know if you have any review comments. thx.

tkol2022 commented on baselines/aad.md:
Thanks for catching this. After meeting with Andrew yesterday, he agreed to update all the front and back matter for the baselines after we complete the formal internal review process. This way each individual author does not have to worry about updating this content and can focus on the policies.

tkol2022 commented:
> I reviewed the content of the AAD baseline on the emerald branch. It is approved from my point of view with these changes:
> 
> * **MS.AAD.7.* controls**: Clarify that Microsoft AAD PIM qualifies as a pam system and that a non-PIM PAM may be used if it meets the same set of requirements.  Terminology from PIM is sprinkled throughout these controls (e.g. permanent active), implying that PIM must be used.   Overall, my suggestion is to be explicit that PIM terminology is used through and that another PAM system may be used assuming it meets the requirements, but also that PIM is acceptable to meet the pam requirement in MS.AAD.7.5v1
> * (nit) **MS.AAD.4.1v1**: please used fixed width font for the log types so that they stand out from the surrounding text

Added the back ticks so that the logs stand out.

tkol2022 commented:
Per discussion at stand-up, removed the policy bullets from Appendix A. Left the reference to the Hybrid Identity Solutions document in tact and fixed the URL since it was wrong.

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
```diff
@@ -497,9 +494,9 @@ User passwords SHALL NOT expire.
+1. [Follow the instructions at this link](https://learn.microsoft.com/en-us/microsoft-365/admin/manage/set-password-expiration-policy?view=o365-worldwide#set-password-expiration-policy) and configure the **Password expiration policy** to **Set passwords to never expire**.
 
 ## 7. Highly Privileged User Access
 
```
Fixed.

tkol2022 commented on baselines/aad.md:
Fixed. That phrase did not belong there anyway because it was describing the exception case, which can be implemented via the config file exclusions in ScubaGear.

tkol2022 commented on baselines/aad.md:
Fixed. Nice catch!!

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
```diff
@@ -586,7 +583,7 @@ User activation of other highly privileged roles SHOULD trigger an alert.
 
```
Fixed.

tkol2022 commented on baselines/aad.md:
Fixed.

tkol2022 commented on baselines/aad.md:
Policy bullets removed per discussion at the standup with CISA. We kept the link to the Hybrid identity document.

tkol2022 commented on baselines/aad.md:
Andrew is going to update all baselines after the formal internal reviews are complete.

tkol2022 commented on baselines/aad.md:
Nice catch. Changed from templates to your suggestion. I also formatted the block quote so that it looks more aligned and I added bold to the data values to help the reader key in on what we are really configuring here.

tkol2022 commented on baselines/aad.md:
I would add this, however I don't really have content for any of the other buttons in the CAP portal either in this compact instruction format so I think referencing the Create button would be out of place. We do have a [section at the top of the baseline](https://github.com/cisagov/ScubaGear/blob/aad-formal-review-emerald/baselines/aad.md#conditional-access-policies) where we talk about initially setting the policies to Report-Only and following Microsoft's guidance to test them before switching them to On. Take a look at that and then let me know if you want me to add more to the instructions. Keep in mind that we have a lot of these so every sentence we add to CAP configuration instructions, will probably need to get multiplied by as many CAP configurations as we have in the document.

tkol2022 commented on baselines/aad.md:
```diff
@@ -69,22 +66,22 @@ Legacy authentication SHALL be blocked.
-2.  Follow [the instructions on this page](https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/howto-conditional-access-policy-block-legacy) to create a conditional access policy that blocks legacy authentication.
+2.  Follow [the instructions on this page](https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/howto-conditional-access-policy-block-legacy) to create a conditional access policy that blocks legacy authentication.
 
 ## 2. Risk Based Policies
```
Fixed.

tkol2022 commented on baselines/aad.md:
```diff
@@ -127,7 +124,7 @@ Sign-ins detected as high risk SHALL be blocked.
-####  MS.AAD.2.1v1, instructions:
+####  MS.AAD.2.1v1 instructions:
 
 1.  Create a conditional access policy that blocks users determined to be high risk by the Identity Protection service.
```
We discussed this offline and agreed upon sticking with the format that I'm using but make some tweaks per Addam's suggestion. I did temporarily create an alternative version of the instructions with step by step 1 2 3 traditional style for comparison and we both agreed that the template version looks nicer. Here's the improved final version:

![image](https://github.com/cisagov/ScubaGear/assets/107076927/02791c04-4d64-407a-bee5-e0a3054ff152)

tkol2022 commented on baselines/aad.md:
```diff
@@ -497,9 +494,9 @@ User passwords SHALL NOT expire.
-1. [Follow the instructions at this link](https://docs.microsoft.com/en-us/microsoft-365/admin/manage/set-password-expiration-policy?view=o365-worldwide#set-password-expiration-policy) and configure the **Password expiration policy** to **Set passwords to never expire**.
+1. [Follow the instructions at this link](https://learn.microsoft.com/en-us/microsoft-365/admin/manage/set-password-expiration-policy?view=o365-worldwide#set-password-expiration-policy) and configure the **Password expiration policy** to **Set passwords to never expire**.
 
 ## 7. Highly Privileged User Access
```
Fixed.

tkol2022 commented on baselines/aad.md:
```diff
@@ -708,7 +705,7 @@ Get-MgUser -Filter "userPrincipalName eq 'username@somedomain.com'" | FL
+#### MS.AAD.7.9v1 instructions:
  
  1. Follow the same instructions as MS.AAD.7.8v1 for each of the highly privileged roles (other than Global Administrator) but enter a security monitoring mailbox that is different from the one used to monitor Global Administrator activations.
 
```
Fixed.

tkol2022 commented:
@gdasher @schrolla @nanda-katikaneni Made the requested revisions. Take a look if you like and kindly perform the GitHub approval if you are satisfied. Thanks for the diligent review.

gdasher commented on baselines/aad.md:
```diff
@@ -23,21 +23,20 @@ or [G3](https://www.microsoft.com/en-us/microsoft-365/government)
-[home tenant](https://docs.microsoft.com/en-us/azure/active-directory/external-identities/authentication-conditional-access) is the one that hosts the guest user’s identity.
+**Resource Tenant & Home Tenant** – In scenarios where [guest users are involved](https://learn.microsoft.com/en-us/azure/active-directory/external-identities/authentication-conditional-access) the **resource tenant** hosts the M365 target resources that the guest user is accessing. The **home tenant** is the one that hosts the guest user's identity.
 
 ## Highly Privileged Roles
```
(nit)

Please list each highly privileged role on a new line.  The current format is strange IMHO.

tkol2022 replied:
Fixed.

gdasher reviewed (approved):
I skimmed through one more time.  Looks good with one small nit.

ssatyapal123 reviewed (changes requested):
Thanks for the opportunity to review. It looks great.
Small suggestion in Policy
**MS.AAD.2.1v1**
Users detected as high risk SHALL be blocked
Rationale: By blocking users determined as high risk, compromised accounts are prevented from accessing the tenant.

Suggest changing Rationale to:
_Rationale: By blocking users determined as high risk, compromised accounts **can be** prevented from accessing the tenant._

tkol2022 commented:
> Thanks for the opportunity to review. It looks great. Small suggestion in Policy **MS.AAD.2.1v1** Users detected as high risk SHALL be blocked Rationale: By blocking users determined as high risk, compromised accounts are prevented from accessing the tenant.
> 
> Suggest changing Rationale to: _Rationale: By blocking users determined as high risk, compromised accounts **can be** prevented from accessing the tenant._

Fixed.

nanda-katikaneni reviewed (approved):
Looks good to me.
